packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -241,12 +240,10 @@ export class DestinationFromServiceRetriever {
     if (destination.originalProperties?.['tokenServiceURLType'] !== 'Common') {
       return undefined;
     }
-    const subdomainSubscriber = getIssuerSubdomain(
-      this.subscriberToken?.userJwt?.decoded
-    );
-    const subdomainProvider = getIssuerSubdomain(
-      this.providerServiceToken?.decoded
-    );
+    const subdomainSubscriber =
+      getSubdomain(this.subscriberToken?.serviceJwt?.decoded) ||
+      getSubdomain(this.subscriberToken?.userJwt?.decoded);
```

Comment:
Previously we only looked at the subdomain from the user JWT. However, if the user passed a separate `iss` property this should be the actual subdomain and it should take precedence (right?).

If the `iss` property was passed, the service JWT was fetched with that subdomain and should have the same subdomain as it is an XSUAA token. If it was not possible to fetch the service token with this subdomain, I assume it wouldn't work in this case either.